oak_channel/src/lib.rs

```diff
@@ -79,19 +80,13 @@ impl InvocationChannel {
     pub fn read_message<M: message::Message>(&mut self) -> anyhow::Result<M> {
         // `encoded_message` will contain the full message we are going to read. Instead of
         // allocating separate buffers and copying data into `encoded_message`, we will ensure that
-        // `encoded_message` has enough space and pass mutable slices pointing into it to the
-        // `read_frame` calls.
-        let mut encoded_message: Vec<u8> = Vec::new();
+        // `encoded_message` has enough capacity. There will be at least a START and an END frame
+        // (4k each), so we start with a reasonable initial capacity.
+        let mut message_buffer = BytesMut::with_capacity(4096);
```

Comment:
This is not accurate. Frames are up to 4k each but can be less. Messages can be just one frame. Imo we should skip assuming a capacity here and remove the comment.

Reply:
Improved the comment, but left the 4096, for the following reason: read_frame will crash when capacity is insufficient (by design). Expectation is that capacity is added at the lib.rs level. I think it should be fine **unless** messages are much smaller than 4k in most cases.